### DISCUSSION TITLE
Creating 2D microstructure with given number of vertices

### DISCUSSION THREAD

**Discussion by kumaranvenk** [neperfepx/neper#763] [Q&A]
Please suggest how to create desired number of vertices in neper. Also, please suggest, if I have dimensions related to vertices, how can I create the microstructure from it?

**Comment by rquey** [comment]
If by "vertices", you mean "cells", then use `-n`:
```
neper -T -n 10
neper -V n10-id1.tess -print img
```
![img](https://github.com/neperfepx/neper/assets/2659376/5dcac17b-931a-49ac-b05d-09ca25482c77)

If "dimension" is the size of the cell, then use `-morpho diameq:...`:
```
neper -T -n 10 -morpho "diameq:lognormal(1,0.3),1-sphericity:lognormal(0.145,0.03)"
neper -V n10-id1.tess -print img
```
![img](https://github.com/neperfepx/neper/assets/2659376/716cf837-aa6c-4d52-9109-88f8983f92a8)

**Reply by kumaranvenk** [reply]
by vertices (or vertex) I meant, where the two or more lines(boundaries) of microstructure meet to form a junction. For eg. When i run a neper code:

neper -T -n 5 -dim 2; it will give me 12 vertices. The coordinates of these points will be available in .tess file. Lets, assume if I want to generate a microstructure with 15 vertex, how we can do. Also if I have coordinates of vertex, how I can use that coordinates to create microstructure.

Btw, thanks for your reply

**Reply by rquey** [reply]
Okay. In that case, Neper does not offer direct control on the vertices.  What would be the interest of this? What is your application?

**Reply by kumaranvenk** [reply]
I have SEM image of Low carbon steel. I want to simulate the data extracted from the SEM image (like, number of microstructure, size of each microstructure) in the Neper (using weighted voronoi algorithm, where the weight will be the equivalent diameter).

Any suggestion, how I can do that??

**Reply by rquey** [reply]
If you can write it as a [tesr](https://neper.info/doc/fileformat.html#raster-tessellation-file-tesr), you can then use the [-stat*](https://neper.info/doc/neper_t.html#post-processing-options) options to extract information (see https://neper.info/doc/tutorials/ebsd_process.html). You can also use [-morpho tesr:...](https://neper.info/doc/neper_t.html#cmdoption-morpho)  to generate a tessellation that "best fits" the EBSD map.
